src/Microsoft.AspNetCore.Proxy/ProxyAdvancedExtensions.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Proxy
+{
+    static class ProxyAdvancedExtentions
```

Comment:
OK. Didn't know that convention. Will fix. Sorry for that. If we are talking about being explicit, why are you never explicit about class being "sealed"? That looks really strange but I see this very often.